akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Protocol.scala

```diff
@@ -98,8 +98,6 @@ private[http] object Http2Protocol {
    */
   final val NoStreamId = 0
 
-  final val PushPromiseEnabledDefault = true
```

Comment:
This is dead code and seems like it can be replaced with `enable-push = on` in reference.conf.

Reply:
This is not dead code. 

A server may send PUSH_PROMISE frames before the client had time to send the SETTINGS.